heap/heap.h

```diff
@@ -0,0 +1,22 @@
+#ifndef DATASTRUCTURES_HEAP_H
+#define DATASTRUCTURES_HEAP_H
+#include <vector>
+
+class Heap {
+    std::vector <int> array;
+
+    void sift_up(int indx);
+    void sift_down(int indx);
+
+public:
+    void push(int el);
+    void pop();
+    int top();
+    void print();
+    Heap(const std:: vector <int> &mass) : array(mass){
```

Comment:
Это вообще не похоже ни на один кодстайл, не ставятся пробелы после разрешения пространства имён (::) и до параметров шаблона (<int>):
```suggestion
    Heap(const std::vector<int> &mass) : array(mass){
```